pkgs/leak_tracker/lib/src/leak_tracking/_object_record.dart

```diff
@@ -97,19 +97,14 @@ class ObjectRecord {
   }
 
   void setContext(String key, Object value) {
-    final theContext = context ?? {};
-    theContext[key] = value;
-    context = theContext;
+    context ??= {};
+    context![key] = value;
   }
 
   void mergeContext(Map<String, dynamic>? addedContext) {
     if (addedContext == null) return;
-    final theContext = context;
-    if (theContext == null) {
-      context = addedContext;
-      return;
-    }
-    theContext.addAll(addedContext);
+    context ??= {};
+    context!.addAll(addedContext);
   }
 
   LeakReport toLeakReport() => LeakReport(
```

Comment:
The intent of the original code to avoid `!` and to use null-safety.
The new code has '!' and thus moves null-safety validation from compiler to developer, making it error prone.
Can you revert changes in this files?

Reply:
sure.

Reply:
I used the ! syntax because the context must not be null in this case.

How about this change? I will submit it if it's acceptable.

```dart
  void setContext(String key, Object value) {
    context ??= {};
    context?[key] = value;
  }

  void mergeContext(Map<String, dynamic>? addedContext) {
    if (addedContext == null) return;
    context ??= {};
    context?.addAll(addedContext);
  }
```

Reply:
Looks good. Thank you.